src/api/spec/controllers/staging/staging_projects_controller_spec.rb

```diff
@@ -0,0 +1,110 @@
+  let(:target_package) { create(:package, name: 'target_package', project: project) }
+  let(:source_package) { create(:package, name: 'source_package', project: source_project) }
+
+  describe 'GET #show' do
+    context 'not existing project' do
+      before do
+        get :show, params: { name: 'does-not-exist', format: :xml }
+      end
+
+      it { expect(response).to have_http_status(:not_found) }
+    end
+
+    context 'valid project' do
+      let(:broken_packages_backend) do
+        <<~XML
+          <resultlist state="e9e2a9641ea83ac9c8786c24ca57dc6f">
+            <result project="home:foo:Staging:A" repository="openSUSE_Tumbleweed" arch="i586" code="published" state="published">
+              <status package="barbar" code="failed" />
+            </result>
+            <result project="home:foo:Staging:A" repository="openSUSE_Tumbleweed" arch="x86_64" code="published" state="published">
+              <status package="barbar" code="failed" />
+            </result>
+          </resultlist>
+        XML
+      end
+      let(:broken_packages_path) { "#{CONFIG['source_url']}/build/#{staging_project.name}/_result?code=failed&code=broken&code=unresolvable" }
+
+      let!(:bs_request) do
+        create(:bs_request_with_submit_action,
+               creator: user,
+               target_project: project.name,
+               target_package: target_package.name,
+               source_project: source_project.name,
+               source_package: source_package.name,
+               staging_project: staging_project)
+      end
+
+      let(:untracked_review) { create(:review, by_project: staging_project) }
+      let!(:untracked_request) do
+        create(:bs_request_with_submit_action,
+               creator: user,
+               target_project: project.name,
+               target_package: target_package.name,
+               source_project: source_project.name,
+               source_package: source_package.name,
+               reviews: [untracked_review])
+      end
+
+      let(:review) { create(:review, by_project: staging_project) }
+      let!(:bs_request_to_review) do
+        create(:bs_request_with_submit_action,
+               creator: user,
+               target_project: project.name,
+               target_package: target_package.name,
+               source_project: source_project.name,
+               source_package: source_package.name,
+               staging_project: staging_project,
+               reviews: [review])
+      end
+
+      let(:missing_review) { create(:review, by_user: user) }
+      let!(:bs_request_missing_review) do
+        create(:bs_request_with_submit_action,
+               creator: user,
+               target_project: project.name,
+               target_package: target_package.name,
+               source_project: source_project.name,
+               source_package: source_package.name,
+               staging_project: staging_project,
```

Comment:
Just an idea... you could store the duplicated parts of the requests in an variable and then use it for each factory, like it's done [here](https://github.com/openSUSE/open-build-service/blob/staging-workflow/src/api/spec/models/staging/staging_project_spec.rb#L30).